src/tls.rs

```diff
@@ -0,0 +1,194 @@
+//! Connection helper.
+use tokio::io::{AsyncRead, AsyncWrite};
+
+use tungstenite::{
+    client::uri_mode, error::Error, handshake::client::Response, protocol::WebSocketConfig,
+};
+
+use crate::{client_async_with_config, domain, IntoClientRequest, WebSocketStream};
+
+pub use crate::stream::MaybeTlsStream;
+
+/// A TLS connector that can be used when establishing TLS connections.
+#[non_exhaustive]
+pub enum TlsConnector {
+    /// Plain (non-TLS) connector.
+    Plain,
+    /// `native-tls` TLS connector.
+    #[cfg(feature = "native-tls")]
+    NativeTls(native_tls_crate::TlsConnector),
+    /// `rustls` TLS connector.
+    #[cfg(feature = "rustls-tls")]
+    Rustls(std::sync::Arc<rustls::ClientConfig>),
+}
+
+mod encryption {
+    #[cfg(feature = "native-tls")]
+    pub mod native_tls {
+        use native_tls_crate::TlsConnector;
```

Comment:
I suggest either the `TlsConnector` -> `Connector` renama above, or importing this as `NativeTlsConnector` - it's confusing that this name refers to different types throughout the file.

Reply:
That is definitely true. I decided to rename the crate's connector to simply `Connector` so this should be fine then.
I can additionally rename the import to `NativeTlsConnector` to reduce confusion if you like?

Reply:
> I decided to rename the crate's connector to simply Connector so this should be fine then.

:+1: 

> I can additionally rename the import to NativeTlsConnector to reduce confusion if you like?

Up to you, `NativeTlsConnector` is nicely explicit. But no longer necessary to rename it due to the other rename.